code/modules/research/experimentor/experimentor.dm

```diff
@@ -0,0 +1,319 @@
+	var/obj/item/unknown_tech/T = locate(href_list["item"]) in src
+	if(..())
+		return
+	usr.set_machine(src)
+
+	var/scantype = href_list["function"]
+	if(href_list["close"])
+		usr << browse(null, "window=experimentor")
+		return
+	else if(scantype == "eject")
+		ejectItem()
+	else if(scantype == "refresh")
+		src.updateUsrDialog()
+	else
+		if(!loaded_item)
+			updateUsrDialog() //Set the interface to unloaded mode
+			to_chat(usr, "<span class='warning'>[src] is not currently loaded!</span>")
+			return
+		//It's harder to raise innovation.
+		var/adjusted_flex_cost = base_flex_cost - (round(T.stability/20,1)-2)
+		if(text2num(scantype) == SCANTYPE_POKE)
+			T.adjuststability(rand(2,10))
+			T.adjustinnovation(0-rand(0,6))
+			T.adjustflexibility(0-rand(adjusted_flex_cost, adjusted_flex_cost + 5))
+			to_chat(usr, "<span class='notice'>Mechanical arms carefully grease and work the [T.name] as lasers prune away excess weight.</span>")
+			playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+		if(text2num(scantype) == SCANTYPE_IRRADIATE)
+			T.adjustinnovation(rand(2,6))
+			T.adjuststability(0-rand(0,4))
+			T.adjustpotency(0-rand(0,4))
+			T.adjustflexibility(0-rand(adjusted_flex_cost, adjusted_flex_cost + 5))
+			to_chat(usr, "<span class='notice'>The E.X.P.E.R.I-MENTOR removes a single part from the [T.name], replacing it with a different one.</span>")
+			playsound(src.loc, 'sound/weapons/gun_interactions/pistol_magin.ogg', 50, 1)
+		if(text2num(scantype) == SCANTYPE_GAS)
+			T.adjustpotency(rand(2,10))
+			T.adjustinnovation(0-rand(0,6))
+			T.adjustflexibility(0-rand(adjusted_flex_cost, adjusted_flex_cost + 5))
+			to_chat(usr, "<span class='notice'>The E.X.P.E.R.I-MENTOR replaces some of the more delicate components in the [T.name] with more rugged equivalents.</span>")
+			playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+		if(text2num(scantype) == SCANTYPE_HEAT)
+			T.adjustpotency(rand(2,8))
+			T.adjuststability(0-rand(0,6))
+			T.adjustflexibility(0-rand(adjusted_flex_cost, adjusted_flex_cost + 5))
+			to_chat(usr, "<span class='notice'>The [T.name] glows ever so slightly as its core clock speed increases.</span>")
+			playsound(loc, "sparks", 75, 1, -1)
+		if(text2num(scantype) == SCANTYPE_COLD)
+			T.adjuststability(rand(2,8))
+			T.adjustpotency(0-rand(0,6))
+			T.adjustflexibility(0-rand(adjusted_flex_cost, adjusted_flex_cost + 5))
+			to_chat(usr, "<span class='notice'>The E.X.P.E.R.I-MENTOR adds a new coolant cell and some tubing to the [T.name].</span>")
+			playsound(loc, 'sound/items/welder.ogg', 50, 1)
+		if(text2num(scantype) == SCANTYPE_DISCOVER)
+			loaded_item = discover(T)
+			ejectItem()
+			to_chat(usr, "<span class='notice'>The E.X.P.E.R.I-MENTOR assembles the [T.name] into its new, final form!</span>")
+			playsound(src.loc, 'sound/items/rped.ogg', 50, 1)
+		if(text2num(scantype) == SCANTYPE_OBLITERATE)
+			T.reroll()
+			T.adjustflexibility(0-rand(adjusted_flex_cost*2+5, adjusted_flex_cost*3+5))
```

Comment:
Seems to be the same as just going 5-rand(adjusted_flex_cost*2, adjusted_flex_coast*3), which reads easier, imo.